pennylane/optimize/qng.py

```diff
@@ -155,19 +156,26 @@ def __init__(self, stepsize=0.01, diag_approx=False, lam=0):
         self.metric_tensor = None
         self.lam = lam
 
-    def step_and_cost(self, qnode, x, recompute_tensor=True, metric_tensor_fn=None):
+    def step_and_cost(
+        self, qnode, *args, grad_fn=None, recompute_tensor=True, metric_tensor_fn=None, **kwargs
+    ):
         """Update the parameter array :math:`x` with one step of the optimizer and return the
         corresponding objective function value prior to the step.
 
         Args:
             qnode (QNode): the QNode for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            *args : variable length argument list for qnode
+            grad_fn (function): optional gradient function of the
+                qnode with respect to the variables ``*args``.
+                If ``None``, the gradient function is computed automatically.
+                Must return the same shape of tuple [array] as the autograd derivative.
             recompute_tensor (bool): Whether or not the metric tensor should
                 be recomputed. If not, the metric tensor from the previous
                 optimization step is used.
             metric_tensor_fn (function): Optional metric tensor function
                 with respect to the variables ``x``.
                 If ``None``, the metric tensor function is computed automatically.
+            **kwargs : variable length of keyword arguments for the qnode
```

Comment:
Good catch. Was this needed because you were using keyword arguments in your notebook?

Reply:
No, this is to make `QNGOptimizer` more similar to `GradientDescentOptimizer` :)